### PR TITLE
Parse `<Time>` attribute from Upload.xml

### PR DESCRIPTION
**Pull request by zackgalbreath**
Teach CDash to use this newly provided data from CTest:

https://gitlab.kitware.com/cmake/cmake/-/merge_requests/9292

### CODE REVIEW

**Comment by zackgalbreath** [comment]
This will need a rebase after the changes from #2047 land in `master`.

File: app/cdash/tests/test_starttimefromupload.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        $file = dirname(__FILE__) . '/data/StartTimeFromUpload/Upload.xml';
+        if (!$this->submission('StartTimeFromUpload', $file)) {
+            $this->fail("Failed to submit {$file}");
+        }
+
+        // No errors in the log.
+        $this->assertTrue($this->checkLog($this->logfilename) !== false);
+
+        // The build exists.
+        $results = DB::select('SELECT id FROM build WHERE projectid = ?', [(int) $this->project->Id]);
```
Comment (williamjallen) [inline app/cdash/tests/test_starttimefromupload.php:47]: You could use Eloquent for this instead:
```suggestion
        $results = Project::findOrFail((int) $this->project->Id)->builds()->get();
```

File: app/cdash/xml_handlers/upload_handler.php [hunk 2]
```diff
@@ -48,6 +48,8 @@ class UploadHandler extends AbstractHandler
     private $Base64TmpFileWriteHandle;
     private $Base64TmpFilename;
     private $Label;
+    private int $Timestamp;
```
Comment (williamjallen) [inline app/cdash/xml_handlers/upload_handler.php:51]: Would using Laravel's [Carbon](https://github.com/briannesbitt/Carbon) wrapper to manipulate these timestamps be realistic?  I know we're using legacy PHP date/time functions elsewhere in this file, so there's certainly an argument to be made for remaining consistent, but using Carbon will allow us to interact with datetimes more effectively once most things are switched to Eloquent.  Eloquent can auto-populate datetime fields in models with Carbon objects.

File: app/cdash/tests/test_starttimefromupload.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        $file = dirname(__FILE__) . '/data/StartTimeFromUpload/Upload.xml';
+        if (!$this->submission('StartTimeFromUpload', $file)) {
+            $this->fail("Failed to submit {$file}");
+        }
+
+        // No errors in the log.
+        $this->assertTrue($this->checkLog($this->logfilename) !== false);
+
+        // The build exists.
+        $results = \App\Models\Project::findOrFail((int) $this->project->Id)->builds()->get();
```
Comment (williamjallen) [inline app/cdash/tests/test_starttimefromupload.php:47]: I believe you could fix the ignored PHPStan error by doing this:
```suggestion
        $results = \App\Models\Project::findOrFail((int) $this->project->Id)->builds()->firstOrFail();
```

Comment (zackgalbreath) [reply app/cdash/tests/test_starttimefromupload.php]: My latest force push got me as close as I could 🤷 . If I change it to `::firstOrFail()` the test fails with

```
Unexpected exception of type [BadMethodCallException] with message [Method Illuminate\Database\Eloquent\Relations\HasMany::firstOrFail does not exist.]
```

Comment (williamjallen) [reply app/cdash/tests/test_starttimefromupload.php]: That looks better to me.  The static/dynamic method duality magic Laravel does confuses PHPStan and I eventually hope to add some rules to specifically ignore Laravel functions, which should resolve the new error.

**Review by williamjallen — APPROVED** [review]

**Pull request merged**